core/compress_api.php

```diff
@@ -83,6 +83,13 @@ function compress_handler_is_enabled() {
  * @access public
  */
 function compress_start_handler() {
+	# Do not start compress handler if we got any output so far, as this
+	# denotes that an error has occurred. Enabling compression in this case
+	# will likely cause a Content Encoding Error when displaying the page.
+	if( ob_get_contents() ) {
```

Comment:
> would expect that checking ( ob_get_length() > 0 ) is better in terms of performance.

Good point, will change.

> Of course, there might be output from require_api( 'compress_api.php' );

That's exactly the behavior I'm trying to address here - any errors/warnings/notices that are thrown as we start initialize the core APIs. This single require_api() call actually includes a dozen other APIs

So I'm afraid this check *is* needed.